browser/ui/brave_browser_command_controller.cc

```diff
@@ -122,8 +124,14 @@ void BraveBrowserCommandController::InitBraveCommandState() {
 #if BUILDFLAG(ENABLE_TOR)
   UpdateCommandForTor();
 #endif
-  UpdateCommandEnabled(IDC_ADD_NEW_PROFILE, !is_guest_session);
-  UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, !is_guest_session);
+  if (!is_guest_session) {
+    PrefService* service = g_browser_process->local_state();
+    DCHECK(service);
+    if (service->GetBoolean(prefs::kBrowserGuestModeEnabled)) {
+      UpdateCommandEnabled(IDC_ADD_NEW_PROFILE, true);
+      UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, true);
```

Comment:
This seems not equivalent to what it was, before it would update to either true or false, and now it will only update to true when all checks passed.
I think it's better to be written like:
```
UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, !is_guest_session && guest_enabled);
```

Reply:
Do you mean:

    UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, !is_guest_session);
    if (!is_guest_session) {
      PrefService* service = g_browser_process->local_state();
      DCHECK(service);
      bool guest_enabled = service->GetBoolean(prefs::kBrowserGuestModeEnabled));
      UpdateCommandEnabled(IDC_ADD_NEW_PROFILE, !is_guest_session && guest_enabled);
      UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, !is_guest_session && guest_enabled);
    }

or

    UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, !is_guest_session);
    bool guest_enabled = false;
    if (!is_guest_session) {
      PrefService* service = g_browser_process->local_state();
      DCHECK(service);
      guest_enabled = service->GetBoolean(prefs::kBrowserGuestModeEnabled));
    }
    UpdateCommandEnabled(IDC_ADD_NEW_PROFILE, !is_guest_session && guest_enabled);
    UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, !is_guest_session && guest_enabled);

Reply:
I meant:
```
PrefService* service = g_browser_process->local_state();
DCHECK(service);

bool guest_enabled = service->GetBoolean(prefs::kBrowserGuestModeEnabled));
UpdateCommandEnabled(IDC_OPEN_GUEST_PROFILE, !is_guest_session && guest_enabled);
```

Reply:
Yes, the last one. :)